gdal/ogr/ogrsf_frmts/dxf/ogrdxflayer.cpp

```diff
@@ -2190,21 +2143,20 @@ OGRDXFFeature *OGRDXFLayer::TranslateSPLINE()
 /*      Use the helper function to check the input data and insert      */
 /*      the spline.                                                     */
 /* -------------------------------------------------------------------- */
-    OGRLineString *poLS = InsertSplineWithChecks( nDegree,
-        adfControlPoints, nControlPoints, adfKnots, nKnots, adfWeights );
+    auto poLS = std::unique_ptr<OGRLineString>(InsertSplineWithChecks( nDegree,
+        adfControlPoints, nControlPoints, adfKnots, nKnots, adfWeights ));
```

Comment:
This function is defined in this file so it would make as much sense to change it to return `unique_ptr`. But maybe you want to do that in another commit.

Reply:
yeah, that can be done later. I just wanted to fix a few things on which cppcheck stumbled upon (false positives), and did similar changes through that file.